charms/reconciler.py

```diff
@@ -29,14 +30,9 @@ def reconcile(self, event):
 
         self.stored.reconciled = False
 
-        with status.context(self.charm.unit):
+        with status.context(self.charm.unit, self.exit_status):
             try:
-                result = self.reconcile_function(event)
+                self.reconcile_function(event)
                 self.stored.reconciled = True
-                return result
-            except Exception as e:
-                # The decision to enter Blocked instead of Error here is going
-                # to be controversial for sure. I think it's worth it to cut
-                # down on try/except boilerplate in reconcile functions.
-                log.exception(e)
-                status.add(BlockedStatus("Failed to reconcile, see debug-log"))
```

Comment:
Hooray!  any uncaught exceptions in a reconciler loop now will set the charm in error status.  If the reconciler throws an exception that's unexpected -- we aren't blocked -- we are in error.